great_tables/_formats.py

```diff
@@ -1771,6 +1771,209 @@ def fmt_time_fn(
+        The default is all rows, resulting in all rows in `columns` being formatted. Alternatively,
+        we can supply a list of row indices.
+    label : str | None
+        The visible 'label' to use for the link. If `None` (the default) the URL will serve as the
+        label. There are two non-`None` options: (1) using a piece of static text for the label
+        through provision of a string, and (2) a function can be provided to fashion a label from
+        every URL.
+    as_button : bool
+        An option to style the link as a button. By default, this is `False`. If this option is
+        chosen then the `button_fill` argument becomes usable.
+    color : str
+        The color used for the resulting link and its underline. This is `"auto"` by default; this
+        allows **Great Tables** to choose an appropriate color based on various factors (such as the
+        background `button_fill` when `as_button` is `True`).
+    show_underline : str | bool
+        Should the link be decorated with an underline? By default this is `"auto"` which means that
+        **Great Tables** will choose `True` when `as_button=False` and `False` in the other case.
+        The link underline will be the same color as that set in the `color` option.
+    button_fill : str
+        The color to fill the button with. This is ignored if `as_button` is `False`. The default is
+        `"auto"` which will use the default background color for the button.
+    button_width : str
+        The width of the button. This is ignored if `as_button` is `False`. The default is `"auto"`
+        which will use the default button width for link-as-button.
+    button_outline : str
+        Options for styling a link-as-button (and only applies if `as_button=True`). This option is
+        by default set to `"auto"`, allowing **Great Tables** to choose the appropriate outline
+        color value.
+
+    Returns
+    -------
+    GT
+        The GT object is returned.
+    """
+
+    if as_button:
+        #
+        # All determinations of `color`, `show_underline`, `button_fill` and
+        # `button_width` for the case where `as_button=True`; each of the
+        # above arguments are set to "auto" by default
+        #
+
+        # In the button case, we opt to never show an underline unless it's
+        # requested by the user (i.e., `show_underline=True`)
+        if show_underline == "auto":
+            show_underline = False
+
+        if button_width == "auto":
+            button_width = None
+
+        button_outline_color = button_outline
+        button_outline_style = "solid"
+        button_outline_width = "2px"
+
+        # There are various combinations of "auto" or not with `button_fill` and
+        # `color` that need to be handled delicately so as to ensure contrast
+        # between foreground text and background fill is maximized
+        if button_fill == "auto" and color == "auto":
```

Comment:
Does this `if` block not need to set button_outline_color? (which could be "auto")